packages/next/src/server/lib/route-resolver.ts

```diff
@@ -168,7 +168,8 @@ export async function makeResolver(
       route.type === 'header' ||
       route.name === 'catchall route' ||
       route.name === 'middleware catchall' ||
-      route.name?.includes('check')
+      (route.name?.includes('check') &&
+        route.name !== 'dynamic route/page check')
```

Comment:
Not sure if this is the right way to fix it, what's supposed to be included/excluded here?